l3kernel/CHANGELOG.md

```diff
@@ -7,18 +7,57 @@ this project uses date-based 'snapshot' version identifiers.
 
 ## [Unreleased]
 
+## [2025-03-10]
+
+### Added
+- `\text_words_map_function:nN`
+- Variants `\tl_head:e` and `\tl_tail:e`
+- `l3tl` functions (and variants) which trim spaces from only one side
+  (issue \#1673):
+  - `\tl_trim_left_spaces:n`, `\tl_trim_left_spaces_apply:nN`,
+    `\tl_(g)trim_left_spaces:N`
+  - `\tl_trim_right_spaces:n`, `\tl_trim_right_spaces_apply:nN`,
+    `\tl_(g)trim_right_spaces:N`
+- Module `l3graphics`, moved from `l3experimental` to stable status
+
+### Changed
+- `\fp_set_function:nnn` and `\fp_set_variable:nn` now raise errors on
+  undefined (fp) identifiers (issue \#1700)
+
+### Fixed
+- Avoid low-level error if keys are given without a module (issue \#1254)
+
+## [2025-01-18]
+
+### Changed
+- Use a dedicated internal marker for `fp` and `int` expressions, allowing use
+  with LuaMetaTeX (see \#1518)
+
+### Fixed
+- Missing trailing period in `\tl_analysis_log:(N|n)` output (see \#1663)
+- Some state variables weren't correctly reset by `\keys_define:nn` and
+    `\keys_set:nn` breaking some nested usages in edge cases
+
+## [2025-01-14]
+
+### Added
+- `\int_step_tokens:nn`, `\int_step_tokens:nnn`, and `\int_step_tokens:nnnn`
+
+### Fixed
+- Check conditionals are defined when creating variants (see \#1189)
+
```

Comment:
As shown here (probably as a result of being open too long) this PR appears to have picked up many unrelated changes It could probably benefit from being rebased. that said I have looked at the files locally with the l3graphics files moved to l3kernel (and backend) and it all looks good so I think I'll approve as is.

Reply:
Thanks - I'll tidy up before a merge